server/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -464,7 +464,7 @@ public boolean isCancelled() {
                 }
             };
         }
-        return scheduler.scheduleWithFixedDelay(new Flush(), flushInterval, ThreadPool.Names.GENERIC);
+        return scheduler.scheduleWithFixedDelay(new Flush(), flushInterval, EsExecutors.DIRECT_EXECUTOR_SERVICE);
```

Comment:
NB the scheduler here is a `ScheduledExecutorServiceScheduler` which in fact ignores the executor parameter and executes the action on the scheduler thread, so `DIRECT_EXECUTOR_SERVICE` is appropriate.